test/Api.Test/Vault/AuthorizationHandlers/GroupAuthorizationHandlerTests.cs

```diff
@@ -26,7 +28,6 @@ public class GroupAuthorizationHandlerTests
         CurrentContextOrganization organization)
     {
         organization.Type = userType;
-        organization.LimitCollectionCreationDeletion = true;
```

Comment:
Continuing on from my last comment about this - you've restored some of the test arrangements, but not all. The concern isn't necessarily flaky tests, it's also the risk that tests pass when they should be failing.

In this case, for example, a user can read all groups if (1) they can create collections or (2) they're an owner or admin. This test is meant to be testing (2), but if this value isn't expressly set to `true` to limit collection creation, it'll always pass because of (1) instead - without effectively testing (2).

(Another side effect of having `true` as a default value instead of `false`! alas)

I think that you need to have a helper method which creates the OrgAbilities object and sets up the mock return on the applicationCache dependency, then replace each deleted line here with a call to that helper method.

Reply:
Hmm - I'm following. Updated tests and cleaned up existing usages of inline dictionary creation via a helper arrange method.

Reply:
Looks good! And looks like you removed it from tests that didn't depend on it, thanks for going through it closely.